components/grade-table.js

```diff
@@ -1,23 +1,56 @@
 class GradeTable {
-  constructor(tableElement){
+  constructor(tableElement, noGradesElement){
   this.tableElement = tableElement;
+  this.deleteGrade = null;
+  this.noGradesElement = noGradesElement;
+  this.renderGradeRow = this.renderGradeRow.bind(this);
   }
   updateGrades(grades){
+    var tableBody = document.querySelector("tbody");
+    while (tableBody.firstChild) {
+      tableBody.removeChild(tableBody.firstChild);
+    }
+
     for(var i = 0; i < grades.length; i++){
-      var trElement = document.createElement("tr");
+      this.renderGradeRow(grades[i], this.deleteGrade);
+    }
+    var pElement = document.querySelector("p");
+    if(grades.length === 0){
+      pElement.classList.remove("d-none");
+    } else {
+      pElement.classList.add("d-none");
+    }
+  }
+  onDeleteClick(deleteGrade){
+    this.deleteGrade = deleteGrade;
+  }
+  renderGradeRow(data, deleteGrade){
+      var tableBody = document.querySelector("tbody");
+      var tableRow = document.createElement("tr");
 
-      var tdElement = document.createElement("td");
-      tdElement.textContent = grades[i].name;
-      var td2Element = document.createElement("td");
-      td2Element.textContent = grades[i].course;
-      var td3Element = document.createElement("td");
-      td3Element.textContent = grades[i].grade;
+      var tableData = document.createElement("td");
+      tableData.textContent = data.name;
+      var tableData2 = document.createElement("td");
+      tableData2.textContent = data.course;
+      var tableData3 = document.createElement("td");
+      tableData3.textContent = data.grade;
+      var tableData4 = document.createElement("td");
+      // tableData4.textContent = data.id;
+      var button = document.createElement("button");
+      button.textContent = "DELETE";
+      button.classList.add('btn');
+      button.classList.add('btn-danger');
+      button.addEventListener("click", function () {
+          deleteGrade(data.id);
+        })
+      tableData4.appendChild(button);
 
-      trElement.append(tdElement);
-      trElement.append(td2Element);
-      trElement.append(td3Element);
-      tableElement.append(trElement);
+      tableRow.append(tableData);
+      tableRow.append(tableData2);
+      tableRow.append(tableData3);
+      tableRow.append(tableData4);
```

Comment:
Bundle this to one line if you are going to use append over appendChild

Reply:
I decided to use the method appendChild so that i wouldnt have one long line, easier to code read.